### PR TITLE
Web console: better handle BigInt math

### PR DESCRIPTION
**Pull request by vogievetsky**
There is an issue that the web console will parse BigInts from the server response but BigInts can be tricky to do math with:

image

They do not like to interact with other numbers.

This issue could affect several places (we are saved by the fact that most numbers coming back for the service views are not bigint sized - 16+ digits) but one particular place where this cropped up is setting `maxSize` for a historical to over `1 PB` or `1000000000000000` (one followed by 15 zeros).

In this PR I added a TypeScript type `NumberLike = number | BigInt` and then used it all over the place where we *might* get a BigInt from the Druid API. TS naturally flags all the arithmetic issues that need an explicit `Number(...)` conversion.

So no we can have petabyte scale servers:

image

This PR has:
- [x] been self-reviewed.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: web-console/src/views/datasource-view/datasource-view.tsx [hunk 1]
```diff
@@ -1141,10 +1142,10 @@ ORDER BY 1`;
                   day_aligned_segments,
                   month_aligned_segments,
                   year_aligned_segments,
-                } = original;
+                } = original as Datasource;
                 const segmentGranularities: string[] = [];
                 if (!num_segments || isNaN(year_aligned_segments)) return '-';
-                if (num_segments - minute_aligned_segments) {
+                if (Number(num_segments) - minute_aligned_segments) {
```
Comment (suneet-s) [inline web-console/src/views/datasource-view/datasource-view.tsx:1148]: should this be cast to `NumberLike` because of line 156?

File: web-console/src/views/datasource-view/datasource-view.tsx [hunk 2]
```diff
@@ -224,7 +225,7 @@ interface RetentionDialogOpenOn {
 
 interface CompactionDialogOpenOn {
   readonly datasource: string;
-  readonly compactionConfig: CompactionConfig;
+  readonly compactionConfig?: CompactionConfig;
```
Comment (suneet-s) [inline web-console/src/views/datasource-view/datasource-view.tsx:228]: Why this change and the ` | undefined` below?

**Review by suneet-s — COMMENTED** [review]
seems ok to me. Is there a test that covers this case? I wouldn't want us to accidentally loose petabyte scale status 😝 

I'm no front-end expert, so pardon my silly questions about typecasting

File: web-console/src/views/datasource-view/datasource-view.tsx (hunk 2 above)
Comment (vogievetsky) [reply web-console/src/views/datasource-view/datasource-view.tsx:228]: Because a CompactionDialog can be open on a datasource without a compaction config (like when one is being created for the first time). This was already being set to `undefined` in some places, this change just makes the issue more explicit.

**Review by suneet-s — APPROVED** [review]
Thanks for the explanation! and tests!

petabyte scale platform - here we come!! 🚀

**Comment by vogievetsky** [comment]
Woop woop! Thank you for the review.
